detect-dead-code.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+
+directories="src tools"
```

Comment:
@dipanjanS there is no 'tools' directory in the repo. Please remove this from the directories' list.

Reply:
Same as above (added in this commit)